content/docs/use-cases/versioning-data-and-model-files/index.md

```diff
@@ -1,30 +1,67 @@
-scripts used to manage ML <abbr>artifacts</abbr> (training data, models, etc.)
-on cloud storage. DVC doesn't require special services, and works with
-on-premises storage (e.g. SSH, NAS) as well as any major cloud storage provider
-(Amazon S3, Microsoft Azure, Google Drive,
-[among others](/doc/command-reference/remote/add#supported-storage-types)).
-
-> For hands-on experience, we recommend following the
+[Version control](https://en.wikipedia.org/wiki/Version_control) has become a
+staple in software engineering because it allows effective collaboration on
+source code. This means having a change history to go back (commits), developing
+features in parallel (branching), assisted merging, peer-reviews (pull
+requests), tagging key revisions, etc. Imagine if we could use these features
+for data modeling!
+
+Unfortunately, versioning tools like [Git](https://git-scm.com/) are designed to
+handle small text files. While other assets can exist in the repository, storage
+itself is not the goal, and its limited by Git hosting services
+[like GitHub](https://docs.github.com/en/github/managing-large-files/what-is-my-disk-quota).
+Traditional storage solutions such as hard drives or NAS, or cloud services like
+Amazon S3 or Google Drive, are much better options for saving and transferring
+large files.
+
+What if we could **combine effective data storage with robust versioning
+features**?
+
+![](/img/model-versioning-diagram.png) _DVC's hybrid versioned storage_
+
+DVC brings the best of both worlds together by replacing the data with small,
+human-readable [metafiles](/doc/user-guide/dvc-files-and-directories) that Git
+can handle. The data itself is <abbr>cached</abbr> locally outside the Git repo,
+and can easily be synchronized with on-premises or cloud storage. But unlike
+other options (like
+[Git-LFS](/doc/user-guide/related-technologies#git-lfs-large-file-storage)),
+[remote storage](/doc/command-reference/remote) is optional: no server setup or
+special services are needed, just the `dvc` command.
+
+## How it looks
+
+Working on data processing code, it's easy to overlook that the initial data may
+also change along the way. If something goes wrong (or for any other reason), it
+may be easy to revisit previous versions of the code with Git. But its behavior
+may still be different with the latest data!
+
+To eliminate this variable and achieve full
+[reproducibility](/doc/start/data-pipelines), we can use `dvc add` to capture
+data snapshots that match code checkpoints:
+
+```dvc
+$ git add cleanup.c
+$ dvc add data/raw
+...
+$ git add data/raw.dvc data/.gitignore
+
+$ git commit -m 'Data cleanup v1'
+```
+
+Now that Git is tracking the code (including a `.dvc` file), and DVC is tracking
+the data, we can repeat the procedure to generate more commits. Use
+`dvc checkout` to go back:
+
+```dvc
+$ git checkout v1
+$ dvc checkout
+M       data\raw
+```
+
+![](/img/versioning.png) _Full project restoration_
```

Comment:
The image may need an update to match the file names mentioned above, or to make it more generic. It's currently:

![image](https://user-images.githubusercontent.com/1477535/92066338-3e3fbb80-ed67-11ea-8248-af83aa3f5d2d.png)